### ISSUE TITLE
Fix false positives for shared-line comments in value-list-comma-newline-after

### ISSUE THREAD

**Issue by AndyOGo** [stylelint/stylelint#2906]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?
Improve rule `value-list-comma-newline-after`.
At the moment I can't put comments after commas, like:

````css
a { background-size: 0, // comment fails here...
      0; }
````

> Which rule, if any, is this issue related to?

`value-list-comma-newline-after`

> What CSS is needed to reproduce this issue?

```css
a { background-size: 0, // comment fails here...
      0; }
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "value-list-comma-newline-after": "always-multi-line"
  }
}
```

> Which version of stylelint are you using?

`8.1.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

stylelint-webpack-plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

```shell
ERROR in 
src/styles/_fonts.scss
  8:37  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
  9:93  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 10:76  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 11:74  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 12:77  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 13:88  ✖  Expected empty line before comment                comment-empty-line-before     
 23:38  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 24:89  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 25:72  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 26:70  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 27:73  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 28:84  ✖  Expected empty line before comment                comment-empty-line-before     
 38:34  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 39:89  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 40:72  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 41:70  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 42:73  ✖  Expected newline after "," in a multi-line list   value-list-comma-newline-after
 43:84  ✖  Expected empty line before comment                comment-empty-line-before
```

**Comment by jeddy3** [comment]
@AndyOGo Thanks for the report and for using the template.

This is reproducible with standard CSS comments too:

```css
a {
  background-size:
    0, /* comment fails here... */
    0;
}
```

Produces:

```
3:6 error Expected newline after "," in a multi-line list (value-list-comma-newline-after)
```

I remember [we dealt with shared line comments](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-list-comma-newline-after/index.js#L35-L61) after commas in _selector_ lists at some point. It should just be the cause of porting this logic (or abstracting it out) to the _value_ list rules.

Labelling as "help wanted" and "bug". @AndyOGo Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) if you'd like to see this issue resolved.

**Comment by ricealexander** [comment]
this is still an issue over a year later.

**Comment by ricealexander** [comment]
@hudochenkov fair enough, my apologies
